evalml/pipelines/components/transformers/preprocessing/stl_decomposer.py

```diff
@@ -58,14 +61,19 @@ def __init__(
             seasonal_smoother += 1
 
         self.forecast_summary = None
+        parameters = {
+            "degree": degree,
+            "period": period,
+            "seasonal_smoother": seasonal_smoother,
+            "time_index": time_index,
+            "series_id": series_id,
```

Comment:
If we're only passing in unstacked data for multiseries I don't think we need to use the series_id column, but I left it there for now since I wasn't sure

Reply:
I think keeping both should be fine.

Reply:
I actually took out all the code supporting the ungrouping by series id, since it is the same iteration process as the unstacked data just changing what is being iterated through (i.e. columns instead of groups) but it would probably be really easy to just call the `unstack_multiseries` function if series id is not None, so let me know if you guys think that would be a good thing to add in

Reply:
After talking it over with @eccabay I think we should keep internal stacking inside of the decomposer if it isn't too much of a lift!